fpa/fastapi/fastapi_add.go

```diff
@@ -23,19 +24,39 @@ var (
 	ErrNotStored     = errors.New("cid not stored")
 )
 
-func (i *Instance) Put(ctx context.Context, c cid.Cid) error {
+func (i *Instance) AddFile(ctx context.Context, reader io.Reader) (*cid.Cid, error) {
 	ar := i.auditor.Start(ctx, i.info.ID.String())
-	ar.Close()
-	if err := i.put(ctx, ar, c); err != nil {
+	defer ar.Close()
+	cid, err := i.addFile(ctx, ar, reader)
+	if err != nil {
+		ar.Errored(err)
+		return nil, err
+	}
+	ar.Success()
+	return cid, nil
+}
+
+func (i *Instance) AddCid(ctx context.Context, c cid.Cid) error {
+	ar := i.auditor.Start(ctx, i.info.ID.String())
+	defer ar.Close()
```

Comment:
Bug fix, pretty sure this should be deferred.

Reply:
Yeah, actually the `auditor` concept is still quite undefined... but since it has a `Close()` better to do this.